plugin/trino-hive/src/main/java/io/trino/plugin/hive/coercions/DecimalCoercers.java

```diff
@@ -228,6 +235,61 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
         }
     }
 
+    public static Function<Block, Block> createDecimalToVarcharCoercer(DecimalType fromType, VarcharType toType)
+    {
+        int lengthLimit = toType.getLength().orElse(VarcharType.UNBOUNDED_LENGTH);
+        if (fromType.isShort()) {
+            return new ShortDecimalToVarcharCoercer(fromType, lengthLimit);
+        }
+        return new LongDecimalToVarcharCoercer(fromType, lengthLimit);
+    }
+
+    private static class ShortDecimalToVarcharCoercer
+            extends TypeCoercer<DecimalType, VarcharType>
+    {
+        private final int lengthLimit;
+
+        protected ShortDecimalToVarcharCoercer(DecimalType fromType, int lengthLimit)
+        {
+            super(fromType, VARCHAR);
```

Comment:
I overlooked that before, but we're passing a wrong type here.
it doesn't matter in practice today, but still not the right thing to do

will push a fixup

Reply:
thanks, no need